internal/database/migration/runner/options.go

```diff
@@ -70,18 +72,18 @@ func desugarRevert(schemaContext schemaContext, operation MigrationOperation) (M
 
 	// Construct a map from migration version to the number of its children that are also applied
 	counts := make(map[int]int, len(schemaVersion.appliedVersions))
-	for _, version := range schemaVersion.appliedVersions {
-		counts[version] = 0
-	}
 	for _, version := range schemaVersion.appliedVersions {
 		definition, ok := definitions.GetByID(version)
 		if !ok {
-			return MigrationOperation{}, errors.Newf("unknown version %d", version)
+			continue
 		}
 
 		for _, parent := range definition.Parents {
-			counts[parent]++
+			counts[parent] = counts[parent] + 1
 		}
+
+Ensure that we have an entry for this definition (but do not modify the count)
+		counts[definition.ID] = counts[definition.ID] + 0
```

Comment:
That's not valid Go code, buddy. Here, let me show you:

```suggestion

		// Ensure that we have an entry for this definition (but do not modify the count)
		counts[definition.ID] = counts[definition.ID] + 0
```

Reply:
:facepalm